examples/template_alignment.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+"""Creating a template for a group of subject using alignment
+===================================================================
+
+In this tutorial, we show how to better predict new contrasts for a target
+subject using many source subjects corresponding contrasts. For this purpose,
+we create and template to which we align the target subject.
+
+We mostly rely on python common packages and on nilearn to handle
+functional data in a clean fashion.
+
+
+To run this example, you must launch IPython via ``ipython
+--matplotlib`` in a terminal, or use ``jupyter-notebook``.
+
+.. contents:: **Contents**
+    :local:
+    :depth: 1
+
+"""
+
+# Retrieving the data
+# -------------------
+# In this example we use the IBC dataset, which include a large number of \
+# different contrasts maps for 12 subjects.
+# We download the images for subjects 1 2, 4, 5, 6 and 7 (or retrieve them \
+# if they were already downloaded).
+# Files is the list of paths for each subjects.
+# df is a dataframe with metadata about each of them.
+# mask is an appropriate nifti image to select the data.
+#
+
+from fmralign.fetch_example_data import fetch_ibc_subjects_contrasts
+files, df, mask = fetch_ibc_subjects_contrasts(
+    ['sub-01', 'sub-02', 'sub-04', 'sub-05', 'sub-06', 'sub-07'])
+
+###############################################################################
+# Defining a masker
+# -----------------
+# Using the mask provided, define a nilearn masker that will be used to \
+# handle relevant data. For more information, visit :
+# http://nilearn.github.io/manipulating_images/masker_objects.html
+#
+
+from nilearn.input_data import NiftiMasker
+masker = NiftiMasker(mask_img=mask)
+mask
+masker.fit()
+
+###############################################################################
+# Split the data
+# ---------------------------------------------
+# For each subject, we will use two 'copies' of data acquired in the same \
+# conditions during two independent sessions with a single varying parameter \
+# of acquisition, the phase encoding being either Antero-posterior(AP) or \
+# Postero-anterior(PA).
+#
+# We infer a template for subjects 1 to 6 for both AP and PA data.
+#
+# For subject 7, we split it in two folds, one to learn alignement, the \
+# other one that we'll try to predict:
+#
+# * target train: AP contrasts for subject 7, used to learn alignment towards \
+#   the template
+# * target test: PA contrasts for subject 7, used as a ground truth to score \
+#   our predictions
+#
+from nilearn.image import concat_imgs
+
+# We make a list of 4D niimgs from our list of list of files containing 3D images
+template_train = []
+for i in range(5):
+    template_train.append(concat_imgs(files[i]))
```

Comment:
you really need to learn concat_imgs ?

Reply:
Yes for now to work with 4D images but will be modified by a PR answering issue #19